### PR TITLE
Github: Support pull private repository from pull request

### PR DESCRIPTION
**Pull request by MartinSadovy**
Solves #559, #750

### CODE REVIEW

File: PHPCI/Model/Build/GithubBuild.php [hunk 1]
```diff
@@ -125,7 +125,7 @@ public function getFileLinkTemplate()
 
         if ($this->getExtra('build_type') == 'pull_request') {
             $matches = array();
-            preg_match('/\/([a-zA-Z0-9_\-]+\/[a-zA-Z0-9_\-]+)/', $this->getExtra('remote_url'), $matches);
+            preg_match('/[\/:]([a-zA-Z0-9_\-]+\/[a-zA-Z0-9_\-]+)/', $this->getExtra('remote_url'), $matches);
```
Comment (dancryer) [inline PHPCI/Model/Build/GithubBuild.php]: What does this particular change do?

**Comment by dancryer** [comment]
Looks good. Would be great if someone else could test as I don't have any private github repos...

File: PHPCI/Model/Build/GithubBuild.php (hunk 1 above)
Comment (MartinSadovy) [inline PHPCI/Model/Build/GithubBuild.php]: webhook:
```json
        "ssh_url": "git@github.com:Block8/PHPCI.git",
        "clone_url": "https://github.com/Block8/PHPCI.git",
```
it is char before `Block8/PHPCI`

**Comment by ViGo5190** [comment]
it's work!

**Comment by LewisW** [comment]
I don't think this completely fixes #750, in my experience the problem is caused by the usage of deploy keys. It appears a deploy key is only valid for the main repository, so when people submit a pull request and it tries to fetch from the forked repository, it cannot. 

The fix is to supply your own private key which has permission the main repository and any forks, but it's a painful road of debugging and digging around to work out this is the fix.

**Pull request closed without merging**